backend/modules/student/routers/student_instances.py

```
@@ -291,3 +398,366 @@ async def get_student_simulation_instance(
         logger.error(f"Error in get_student_simulation_instance: {e!r}", exc_info=True)
         raise HTTPException(status_code=500, detail=f"Failed to fetch simulation instance: {e!s}") from e
 
+
+@router.get("/assignment/{assignment_id}/instances", response_model=List[dict])
+async def get_assignment_instances(
+    assignment_id: int,
+    current_user: User = Depends(get_current_user),
```

Ruff (0.14.8)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)